C++: allow to configure bundled translation (#6824)
slint-ui/slint: api/cpp/docs/cmake_reference.md

A scale factor specified this way will also be used to pre-scale images and glyphs when used in combination
with [Resource Embedding](#resource-embedding).

## Bundle translations

> Translations can either be done using `gettext` at runtime, or by bundling all the translated string
> directly into the binary (by putting them in the generated C++ code).

Reviewer: ```suggestion
Translations can either be done using `gettext` at runtime, or by bundling all the translated strings
directly into the binary, by embedding them in the generated C++ code.
```

> If you want to bundle the translation, you need to set the `SLINT_BUNDLE_TRANSLATION` target property
>  to the path in which the translations are. The translation must be in the gettext `.po` format.

Reviewer: ```suggestion
If you want to bundle translations, you need to set the `SLINT_BUNDLE_TRANSLATIONS` target property
to point to a directory containing translations. The translations must be in the gettext `.po` format.
```

> In the following example, the translation file will be looked in `lang/<lang>/LC_MESSAGES/my_application.po`

Reviewer: ```suggestion
In the following example, the translation files will be bundled from `lang/<lang>/LC_MESSAGES/my_application.po`
```

```cmake

> set_property(TARGET my_application PROPERTY SLINT_BUNDLE_TRANSLATION "${CMAKE_CURRENT_SOURCE_DIR}/lang")

Reviewer: ```suggestion
set_property(TARGET my_application PROPERTY SLINT_BUNDLE_TRANSLATIONS "${CMAKE_CURRENT_SOURCE_DIR}/lang")
```
